crates/bevy_pbr/src/material_bind_groups.rs

```diff
@@ -1319,14 +1344,21 @@ where
         ] {
             match self.samplers.get(&bindless_resource_type) {
                 Some(sampler_bindless_binding_array) => {
-                    let sampler_bindings = sampler_bindless_binding_array
+                    let mut sampler_bindings: Vec<_> = sampler_bindless_binding_array
                         .bindings
                         .iter()
                         .map(|maybe_bindless_binding| match *maybe_bindless_binding {
                             Some(ref bindless_binding) => &bindless_binding.resource,
                             None => &**fallback_sampler,
                         })
                         .collect();
+
+                    if let Some(required_binding_array_size) = required_binding_array_size {
+                        while sampler_bindings.len() < required_binding_array_size as usize {
+                            sampler_bindings.push(&**fallback_sampler);
```

Comment:
Using `vec.extend(iter::repeat(fallback).take(n))` will be a little bit clearer here imo.

Reply:
Did the PR work for you?

Reply:
No, see below comment.

Reply:
BTW your suggestion isn't quite correct, but I can do what I think you meant.